### ISSUE TITLE
swift 6 compatablility

### ISSUE THREAD

**Issue by bukira** [Quick/Quick#1302]
Will Quick answer Nimble be updated to support Swift 6 with example of how yo use them?

**Comment by younata** [comment]
It's on my roadmap, but I've been a bit busy lately & haven't found time to get into this as much as I'd like.
For Quick, I fear that supporting the swift 6 language mode is going to be quite hard.
For Nimble, it should be much easier, but still a lot of work.

The work I've done so far has been more for Nimble than Quick.

For now, use the Swift 5 language mode for your Quick & Nimble tests. My apologies.

**Comment by bukira** [comment]
thank you for this, good to know, I shall leave this issue active to others can see it has been asked and answered by you

**Comment by IMCHO** [comment]
It seems that this project and Nimble don't have concurrency problems with `swift-tool-version` 6.0.
You can update package files with no harm, I think.

**Comment by younata** [comment]
> It seems that this project and Nimble don't have concurrency problems with swift-tool-version 6.0.
> You can update package files with no harm, I think.
> If I'm wrong, let me know.

Yes, you can use swift-tools-version 6.0 in your Package.swift, however, you will still need to specify Swift 5 language mode for your unit test targets. Your main targets can still specify Swift 6 language mode, and you can get concurrency checking. I highly recommend turning on Swift 6 language mode for your non-testing targets.
